ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,23 +1,34 @@
 """Serialization Strategy."""
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict
 
 from google.protobuf import type_pb2
-from google.protobuf.internal import decoder
-from google.protobuf.internal import encoder
+from google.protobuf.descriptor import FieldDescriptor
+from google.protobuf.internal import decoder, encoder
+from typing_extensions import Buffer
 
+from ni_measurementlink_service._internal.parameter import _message
+from ni_measurementlink_service._internal.stubs.ni.protobuf.types import xydata_pb2
 
-def _scalar_encoder(encoder) -> Callable[[int], Callable]:
+LowLevelEncoder = Callable[[Callable[[Buffer], int], Any, None], None]
+EncoderFactory = Callable[[int], LowLevelEncoder]
+LowLevelDecoder = Callable[
+    [memoryview, int, int, type_pb2.Field.Kind.ValueType, Dict[FieldDescriptor, Any]], int
+]
+DecoderFactory = Callable[[int, str], LowLevelDecoder]
+
+
+def _scalar_encoder(encoder) -> EncoderFactory:
```

Comment:
It would be helpful to annotate the parameter types, particularly `encoder`, which could be anything.

Also nested functions.